components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -1847,6 +1852,24 @@ private void validateDisplayName(String displayName) throws IdentityOAuthClientE
                     Oauth2ScopeConstants.ErrorMessages.ERROR_CODE_BAD_REQUEST_SCOPE_DISPLAY_NAME_NOT_SPECIFIED
                             .getMessage());
         }
+        if (displayName.length() > Oauth2ScopeConstants.MAX_LENGTH_OF_SCOPE_DISPLAY_NAME) {
+            throw handleClientError(INVALID_REQUEST, String.format(Oauth2ScopeConstants.ErrorMessages.
+                    ERROR_CODE_BAD_REQUEST_SCOPE_DISPLAY_NAME_TOO_LONG.getMessage(), displayName));
+        }
+    }
+
+    /**
+     * Check whether the description is too long.
+     *
+     * @param description Description.
+     * @throws IdentityOAuth2ScopeClientException
+     */
+    private void validateDescription(String description) throws IdentityOAuthClientException {
+
+        if (description.length() > Oauth2ScopeConstants.MAX_LENGTH_OF_SCOPE_DESCRIPTION) {
```

Comment:
Have we validated whether the description is null?

Reply:
Description is not mandatory. So it can be null

Reply:
In that case `description.length()` will throw an NPE right?

Reply:
Fixed